rotkehlchen/chain/aggregator.py

```diff
@@ -1286,50 +1286,70 @@ def get_evm_manager(
+                continue
 
-        If progress_handler is given then provide updates to the frontend about the progress.
+            active_chains.append(chain)
 
-        Returns a list of tuples of the address and the chain it was active in
+        return active_chains
 
-        May raise:
-        - RemoteError if any of the calls fail
-        """
-        filtered_accounts = []
-        for account in accounts:
-            if progress_handler is not None:
-                progress_handler.new_step(f'Checking {account} EVM chain activity')
-            chains: list[SUPPORTED_EVM_CHAINS]
-            if self.ethereum.node_inquirer.get_code(account) != '0x':
-                log.debug(f'Not adding {account} to all evm chains as it is a mainnet contract')
-                chains = [SupportedBlockchain.ETHEREUM]
-            else:  # weird that mypy does not regognize the get_args here properly
-                chains = get_args(SUPPORTED_EVM_CHAINS)  # type: ignore[assignment]
-
-            for chain in chains:
-                chain_manager = self.get_chain_manager(chain)
-                if chain == SupportedBlockchain.AVALANCHE:
-                    # just check balance and nonce in avalanche
-                    has_activity = (
-                        chain_manager.w3.eth.get_transaction_count(account) != 0 or
-                        chain_manager.get_avax_balance(account) != ZERO
-                    )
-                    if has_activity is False:
-                        continue
-                elif chain == SupportedBlockchain.OPTIMISM and chain_manager.node_inquirer.etherscan.has_activity(account) is False:  # noqa: E501
-                    continue
-                # else can only be ethereum so add it unconditionally
-                filtered_accounts.append((chain, account))
+    def track_evm_address(
+            self,
+            address: ChecksumEvmAddress,
+            chains: list[SUPPORTED_EVM_CHAINS],
+    ) -> list[SUPPORTED_EVM_CHAINS]:
+        added_chains = []
+        for chain in chains:
+            try:
+                self.modify_blockchain_accounts(
+                    blockchain=chain,
+                    accounts=[address],
+                    append_or_remove='append',
+                )
+            except InputError:
+                log.debug(f'Not adding {address} to {chain} since it already exists')
+                continue
 
-        return filtered_accounts
+            added_chains.append(chain)
+
+        return added_chains
+
+    def check_chains_and_add_accounts(
+            self,
+            account: ChecksumEvmAddress,
+            chains: list[SUPPORTED_EVM_CHAINS],
+    ) -> list[tuple[SUPPORTED_EVM_CHAINS, ChecksumEvmAddress]]:
+        active_chains = self.check_single_address_activity(
+            address=account,
+            chains=chains,
+        )
+        added_chains = self.track_evm_address(account, active_chains)
```

Comment:
added_chains was confusing to me since I wasn't sure what to expect here. Maybe you can use `new_tracked_chains`?